src/js/containers/aboutTheData/AgencyDetailsContainer.jsx

```diff
@@ -282,38 +123,89 @@ const AgencyDetailsContainer = ({ modalClick, agencyName }) => {
-            recent_update: recentUpdate,
-            discrepancy_count: missingTasCount,
-            obligation_difference: obligationDiff,
-            unlinked_cont_awd: unlinkedCont,
-            unlinked_asst_awd: unlinkedAsst,
-            assurance_statement_file: assuranceStatementFile,
-            tas_account_discrepancies_totals: tasTotals
-        }) => [
-            (<div className="generic-cell-content">{ fiscalYear }</div>),
-            (<div className="generic-cell-content">{ total }</div>),
-            (<CellWithModal data={recentUpdate} openModal={modalClick} modalType="publicationDates" agencyData={{ agencyName }} />),
-            (<CellWithModal data={missingTasCount} openModal={modalClick} modalType="missingAccountBalance" agencyData={{ agencyName, gtasObligationTotal: tasTotals.gtas_obligation_total }} />),
-            (<CellWithModal data={obligationDiff} openModal={modalClick} modalType="reportingDifferences" agencyData={{ agencyName }} />),
-            (<div className="generic-cell-content">{ unlinkedCont }</div>),
-            (<div className="generic-cell-content">{ unlinkedAsst }</div>),
-            (<div className="generic-cell-content"><AgencyDownloadLinkCell file={assuranceStatementFile} /></div>)
-        ]
+    const parseRows = (results, federalBudget) => (
+        results.map((row) => {
+            const rowData = Object.create(BaseReportingPeriodRow);
+            rowData.populate(row, federalBudget);
+            return ([
+                (<div className="generic-cell-content">{ rowData.reportingPeriod }</div>),
+                (<div className="generic-cell-content">{ rowData.percentOfBudget }</div>),
+                (<CellWithModal data={rowData.mostRecentUpdate} openModal={modalClick} modalType="publicationDates" agencyName={agencyName} />),
+                (<CellWithModal data={rowData.missingTAS} openModal={modalClick} modalType="missingAccountBalance" agencyName={agencyName} />),
+                (<div className="generic-cell-content">{ rowData.obligationDifference }</div>),
+                (<div className="generic-cell-content">--</div>),
+                (<div className="generic-cell-content">--</div>),
+                (<div className="generic-cell-content"><AgencyDownloadLinkCell file="placeholder" /></div>)
+            ]);
+        })
     );
 
+    const fetchTableData = useCallback(() => {
+        if (tableRequest.current) {
+            tableRequest.current.cancel();
+        }
+        setLoading(true);
+        const params = {
+            agencyCode,
+            limit: pageSize,
+            page: currentPage,
+            sort: sortStatus.field,
+            order: sortStatus.direction
+        };
+        tableRequest.current = fetchAgency(params);
+        tableRequest.current.promise
+            .then((res) => {
+                // TODO - remove mock federal budget
+                setRows(parseRows(res.data.results, 10000000000000));
```

Comment:
Wasn't sure if the federal budget is already requested / stored in a branch being developed concurrently. I'm guessing it will be most efficient to pass this value as a prop or get it from redux instead of making a duplicate API request?

Reply:
Yeah, I am not sure we have a ticket to capture this. The update to the mocks was after the fact and not sure we scoped work for it yet.

Reply:
@ebdabbs im making the request in `AgenciesContainer`. The value will live in `redux.aboutTheData.totals`.

Reply:
Lol, I was just working on a branch for it :( 

Thank you @MaxwellKendall